### PR TITLE
Updates for contract instance related XDR changes.

### PR DESCRIPTION
**Pull request by dmkozh**
### What

This includes introducing `ContractInstance` instead of `ContractExecutable` and removing some redundant host objects.

### Why

Preparation for introducing contract instance storage

### Known limitations

N/A

### CODE REVIEW

File: soroban-env-host/src/host/declared_size.rs [hunk 1]
```diff
@@ -278,17 +275,17 @@ mod test {
         // xdr types
         expect!["8"].assert_eq(size_of::<TimePoint>().to_string().as_str());
         expect!["8"].assert_eq(size_of::<Duration>().to_string().as_str());
-        expect!["40"].assert_eq(size_of::<ScVal>().to_string().as_str());
-        expect!["40"].assert_eq(size_of::<ScValObject>().to_string().as_str());
-        expect!["80"].assert_eq(size_of::<ScMapEntry>().to_string().as_str());
+        expect!["64"].assert_eq(size_of::<ScVal>().to_string().as_str());
```
Comment (sisuresh) [inline soroban-env-host/src/host/declared_size.rs:278]: Is ScVal bigger due to the instance?

Comment (dmkozh) [reply soroban-env-host/src/host/declared_size.rs:278]: Yes, this is an unfortunate side effect of including it in `ScVal`. FWIW the only reason for it to be `ScVal` in the first place is to be able to store it in `ContractData` LE - but probably it's too late to change its structure.

**Review by sisuresh — APPROVED** [review]

File: soroban-env-host/src/host/declared_size.rs (hunk 1 above)
Comment (sisuresh) [reply soroban-env-host/src/host/declared_size.rs:278]: Yeah that's not great, but I agree too late to mess with this now.

File: soroban-env-common/src/val.rs [hunk 2]
```diff
@@ -323,8 +315,6 @@ impl<E: Env> Compare<Bool> for E {
 // Declare a few extra object wrapper types that don't live anywhere else.
 declare_tag_based_object_wrapper!(VecObject);
 declare_tag_based_object_wrapper!(MapObject);
-declare_tag_based_object_wrapper!(ContractExecutableObject);
```
Comment (jayz22) [inline soroban-env-common/src/val.rs:326]: Why is `ContractExecutableObject` removed as a host object type?

Comment (jayz22) [reply soroban-env-common/src/val.rs:326]: > we just don't need it (so there is not object for contract instance)

Yeah this is what I wasn't sure. So the only place `ContractInstance` exist is in the storage in the xdr form, and there is no need for it as a host object?

Comment (dmkozh) [reply soroban-env-common/src/val.rs:326]: Yes, and I wish it wasn't even in `ScVal`, but that would require restructuring `ContractDataEntry` or introducing a new LE type.

Comment (dmkozh) [reply soroban-env-common/src/val.rs:326]: Actually, we could make this a pointer - that would decrease the ScVal size at cost of slightly increasing the instance entries only, WDYT @sisuresh @jayz22 ?

**Pull request merged**